native/native_main.py

```diff
@@ -118,6 +130,197 @@ def sanitizeFilename(fn):
     return fn
 
 
+def is_valid_firefox_profile(profile_dir):
+    is_valid = False
+    validity_indicator = "times.json"
+
+    if pathlib.WindowsPath(profile_dir).is_dir():
+        test_path = "%s\\%s" % (profile_dir, validity_indicator)
+
+        if pathlib.WindowsPath(test_path).is_file():
+            is_valid = True
+
+    return is_valid
+
+
+def win_firefox_restart(message):
+    """Handle 'win_firefox_restart' message."""
+    reply = {}
+    profile_dir = message["profiledir"].strip()
+
+    ff_bin_name = "firefox.exe"
```

Comment:
We already have a setting for this used for `nativeopen`: `set browser`. It's for people using e.g. Nightly :)

Reply:
Hi @bovine3dom,

Correct me if I'm mistaken, but I don't think the _binary_ is named differently across different Firefox versions _e.g._ Stable, Beta, Developer, Nightly _etc._ all have `Firefox.exe`. The installation path is different across different versions, but the binary name remains the same AFAIK. `ff_bin_name = "firefox.exe"` above will use whichever `Firefox.exe` comes in `%PATH%` first, if the user has multiple versions of Firefox installed.

Are you saying that we should use the value found in `set browser` to restart Firefox with? For example, if the user set `:set browser c:\Program Files\Firefox Developer Edition\Firefox.exe`, `native_main.py` should receive that value when `:restart` is invoked, and use it during the actual restart? 

Please confirm, if the above is what you meant. I can send another patch to handle this. Thanks.

Reply:
Yes, I think so. It would be confusing if I `restart` Nightly and then get stable back, especially if I have set `firefox` to something else.